dev-java/protobuf-java/protobuf-java-4.31.0.ebuild

```diff
@@ -0,0 +1,166 @@
+# Copyright 2008-2025 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# NOTE from https://github.com/protocolbuffers/protobuf/blob/main/cmake/dependencies.cmake
+ABSEIL_MIN_VER="20250127.0"
+
+JAVA_PKG_IUSE="doc source test"
+MAVEN_ID="com.google.protobuf:protobuf-java:${PV}"
+JAVA_TESTING_FRAMEWORKS="junit-4"
+
+inherit java-pkg-2 java-pkg-simple cmake
+
+DESCRIPTION="Core Protocol Buffers library"
+HOMEPAGE="https://protobuf.dev"
+# Currently we bundle the binary version of truth.jar used only for tests, we don't install it.
+# And we build artifact x.y.z from the y.z tarball in order to allow sharing the tarball with
+# dev-libs/protobuf.
+MY_PV="$(ver_cut 2-3)"
+MY_PV="${MY_PV/_rc/-rc}"
+TV="1.4.4"
+SRC_URI="
+	https://github.com/protocolbuffers/protobuf/releases/download/v${MY_PV}/protobuf-${MY_PV}.tar.gz
+	test? (
+		https://repo1.maven.org/maven2/com/google/truth/truth/${TV}/truth-${TV}.jar
+	)
+"
+S="${WORKDIR}/protobuf-${MY_PV}"
+
+LICENSE="BSD"
+SLOT="0/$(ver_cut 1)"
+KEYWORDS="~amd64 ~arm64 ~ppc64 ~amd64-linux ~x86-linux ~x64-macos"
+IUSE="system-protoc"
+
+BDEPEND="
+	system-protoc? ( dev-libs/protobuf:0/${MY_PV}.0[protoc] )
+	!system-protoc? ( >=dev-cpp/abseil-cpp-${ABSEIL_MIN_VER}:= )
+"
+DEPEND="
+	>=virtual/jdk-1.8:*
+	test? (
+		dev-java/guava:0
+		dev-java/mockito:4
+	)
+"
+RDEPEND="
+	>=virtual/jre-1.8:*
+"
+
+JAVA_AUTOMATIC_MODULE_NAME="com.google.protobuf"
+JAVA_JAR_FILENAME="protobuf.jar"
+JAVA_RESOURCE_DIRS="java/core/src/main/resources"
+JAVA_SRC_DIR="java/core/src/main/java"
+
+JAVA_TEST_GENTOO_CLASSPATH="guava,junit-4,mockito-4"
+JAVA_TEST_SRC_DIR="java/core/src/test/java"
+
+run-protoc() {
+	if use system-protoc; then
```

Comment:
This was a hack for cases where `protoc` built from the same source tarball was not available. 
Since this time you are adding the dev-libs in the same pull request, the whole hack should be dropped.